src/main/cpp/Elevator/FeedforwardPID.cpp

```diff
@@ -0,0 +1,272 @@
+}
+
+/**
+ * @brief Starts the timer
+ *
+ */
+void FeedforwardPID::start()
+{
+    timer.Start();
+    isRunning = true;
+}
+
+/**
+ * @brief Stops the timer
+ *  
+ */
+void FeedforwardPID::stop() {
+    timer.Stop();
+}
+
+/**
+ * Resets and starts the timer 
+ * 
+ */
+void FeedforwardPID::reset() {
+    timer.Reset();
+    start();
+}
+
+/**
+ * @brief A feedforward function that gets the elevator pose based on current time
+ *
+ * @param time current system time
+ * @return pose a pose containing the distance
+ */
+Poses::Pose1D FeedforwardPID::getExpectedPose(double time)
+{
+    Poses::Pose1D pose;
+
+    // the time spent accelerating (or decelerating)
+    double acceleration_time = max_velocity / max_acceleration;
+    frc::SmartDashboard::PutNumber("acc time: ", acceleration_time);
+
+    // the time spent maintaining a constant velocity
+    double velocity_time = (max_distance_ - max_velocity * acceleration_time) / max_velocity;
+
+    // velocity should never be negative
+    velocity_time = velocity_time < 0 ? 0.0 : velocity_time;
+    frc::SmartDashboard::PutNumber("vel time: ", velocity_time);
+
+    double reversed_coefficient = reversed ? -1.0 : 1.0;
+
+    frc::SmartDashboard::PutBoolean("phase 1", false);
+    frc::SmartDashboard::PutBoolean("phase 2", false);
+    frc::SmartDashboard::PutBoolean("phase 3", false);
+
+    // if in the acceleration phase
+    if (0 < time && time < acceleration_time)
+    {
+        frc::SmartDashboard::PutBoolean("phase 1", true);
+        pose.acceleration = reversed_coefficient * max_acceleration;
+        pose.velocity = max_acceleration * time;
+        pose.position = 0.5 * pose.velocity * time;
+    }
+
+    // if in the velocity phase
+    else if (velocity_time != 0 && time < acceleration_time + velocity_time)
+    {
+        frc::SmartDashboard::PutBoolean("phase 2", true);
+        pose.acceleration = 0.0;
+        pose.velocity = reversed_coefficient * max_velocity;
+        // adds phase 1 to however much of phase 2 has been gone through
+        pose.position = 0.5 * max_velocity * acceleration_time + max_velocity * (time - acceleration_time);
+    }
+
+    // if in the deceleration phase
+    else if (time < velocity_time + acceleration_time * 2)
+    {
+        frc::SmartDashboard::PutBoolean("phase 3", true);
+
+        double first_phase_distance = 0.5 * max_velocity * acceleration_time;
+        double second_phase_distance = max_velocity * velocity_time;
+        double time_in_triangle = time - (acceleration_time + velocity_time);
+
+        pose.acceleration = -1.0 * reversed_coefficient * max_acceleration;
+        pose.velocity = reversed_coefficient * max_velocity - (reversed_coefficient * max_acceleration * (time_in_triangle));
+
+        // trapezoidal area
+        double third_phase_distance = (max_velocity + pose.velocity) / 2.0 * time_in_triangle;
+
+        pose.position = first_phase_distance + second_phase_distance + third_phase_distance;
```

Comment:
foregt to multiply by reversed for pos, vel, and acc for the entirety of this entire if statement block. You have done some of it, but clearly not all.